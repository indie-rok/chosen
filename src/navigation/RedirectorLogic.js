import React from 'react';
import { Alert } from 'react-native';
import { connect } from 'react-redux';
import { Container, Content, Spinner } from 'native-base';

import { getUser } from '../selectors/user';

export const mapStateToProps = (state) => {
  return {
    user: getUser(state),
  };
};

export class RedirectorLogic extends React.Component {
  componentDidMount() {
    const { navigation, user } = this.props;
    if (user) {
      Alert.alert('Welcome user');
      navigation.replace('ConnectFacebook');
    } else {
      navigation.replace('ConnectFacebook');
    }
  }

  render() {
    return (
      <Container>
        <Content>
          <Spinner color="blue" />
        </Content>
      </Container>
    );
  }
}

export default connect(mapStateToProps)(RedirectorLogic);
