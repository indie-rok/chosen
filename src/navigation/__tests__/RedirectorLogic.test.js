import React from 'react';
import { mount } from 'enzyme';

import { RedirectorLogic, mapStateToProps } from '../RedirectorLogic';
import NavigatorMock from '../../lib/mocks/navigator';

describe('#mapStateToProps', () => {
  it('should return the user from state', () => {
    const state = { user: { email: 'test@test.com' } };
    expect(mapStateToProps(state)).toEqual({
      user: {
        email: 'test@test.com',
      },
    });
  });
});

describe('#RedirectorLogic', () => {
  describe('#componentDidMount', () => {
    it('should redirect to to select pictures if existing user', () => {
      mount(<RedirectorLogic navigation={NavigatorMock} user={{ email: 'test@test.com' }} />);
      expect(NavigatorMock.replace).toHaveBeenCalledWith('SelectPictures');
    });

    it('should redirect to Connect facebook if not existing user', () => {
      mount(<RedirectorLogic navigation={NavigatorMock} user={null} />);
      expect(NavigatorMock.replace).toHaveBeenCalledWith('ConnectFacebook');
    });
  });

  describe('#render', () => {
    it('should mach the snapshot', () => {
      const container = mount(<RedirectorLogic navigation={NavigatorMock} user={null} />);
      expect(container).toMatchSnapshot();
    });
  });
});
