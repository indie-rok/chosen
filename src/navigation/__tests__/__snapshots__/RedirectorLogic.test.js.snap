// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#RedirectorLogic #render should mach the snapshot 1`] = `
<RedirectorLogic
  navigation={
    Object {
      "navigate": [MockFunction],
      "replace": [MockFunction] {
        "calls": Array [
          Array [
            "ConnectFacebook",
          ],
          Array [
            "ConnectFacebook",
          ],
          Array [
            "ConnectFacebook",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
  user={null}
>
  <Styled(Container)>
    <Container
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "height": 1334,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "height": 1334,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
              "height": 1334,
            }
          }
        >
          <Styled(Content)>
            <Content
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                }
              }
            >
              <KeyboardAwareScrollViewMock
                automaticallyAdjustContentInsets={false}
                contentContainerStyle={
                  Array [
                    Object {
                      "padding": undefined,
                    },
                    undefined,
                  ]
                }
                enableAutomaticScroll={true}
                enableOnAndroid={false}
                enableResetScrollToCoords={true}
                extraHeight={75}
                extraScrollHeight={0}
                keyboardOpeningTime={250}
                keyboardShouldPersistTaps="handled"
                resetScrollToCoords={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1,
                  }
                }
                viewIsInsideTabBar={false}
              >
                <ScrollViewMock
                  automaticallyAdjustContentInsets={false}
                  contentContainerStyle={
                    Array [
                      Object {
                        "padding": undefined,
                      },
                      undefined,
                    ]
                  }
                  contentInset={
                    Object {
                      "bottom": 0,
                    }
                  }
                  enableAutomaticScroll={true}
                  enableOnAndroid={false}
                  enableResetScrollToCoords={true}
                  extraHeight={75}
                  extraScrollHeight={0}
                  getScrollResponder={[Function]}
                  handleOnScroll={[Function]}
                  keyboardDismissMode="interactive"
                  keyboardOpeningTime={250}
                  keyboardShouldPersistTaps="handled"
                  keyboardSpace={0}
                  onScroll={[Function]}
                  resetKeyboardSpace={[Function]}
                  resetScrollToCoords={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  scrollEventThrottle={1}
                  scrollForExtraHeightOnAndroid={[Function]}
                  scrollIntoView={[Function]}
                  scrollToEnd={[Function]}
                  scrollToFocusedInput={[Function]}
                  scrollToPosition={[Function]}
                  showsVerticalScrollIndicator={true}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "flex": 1,
                    }
                  }
                  update={[Function]}
                  viewIsInsideTabBar={false}
                >
                  <_class
                    automaticallyAdjustContentInsets={false}
                    contentContainerStyle={
                      Array [
                        Object {
                          "padding": undefined,
                        },
                        undefined,
                      ]
                    }
                    contentInset={
                      Object {
                        "bottom": 0,
                      }
                    }
                    enableAutomaticScroll={true}
                    enableOnAndroid={false}
                    enableResetScrollToCoords={true}
                    extraHeight={75}
                    extraScrollHeight={0}
                    getScrollResponder={[Function]}
                    handleOnScroll={[Function]}
                    keyboardDismissMode="interactive"
                    keyboardOpeningTime={250}
                    keyboardShouldPersistTaps="handled"
                    keyboardSpace={0}
                    onScroll={[Function]}
                    resetKeyboardSpace={[Function]}
                    resetScrollToCoords={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    scrollEventThrottle={1}
                    scrollForExtraHeightOnAndroid={[Function]}
                    scrollIntoView={[Function]}
                    scrollToEnd={[Function]}
                    scrollToFocusedInput={[Function]}
                    scrollToPosition={[Function]}
                    showsVerticalScrollIndicator={true}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "flex": 1,
                      }
                    }
                    update={[Function]}
                    viewIsInsideTabBar={false}
                  >
                    <RCTScrollView
                      automaticallyAdjustContentInsets={false}
                      contentContainerStyle={
                        Array [
                          Object {
                            "padding": undefined,
                          },
                          undefined,
                        ]
                      }
                      contentInset={
                        Object {
                          "bottom": 0,
                        }
                      }
                      enableAutomaticScroll={true}
                      enableOnAndroid={false}
                      enableResetScrollToCoords={true}
                      extraHeight={75}
                      extraScrollHeight={0}
                      getScrollResponder={[Function]}
                      handleOnScroll={[Function]}
                      keyboardDismissMode="interactive"
                      keyboardOpeningTime={250}
                      keyboardShouldPersistTaps="handled"
                      keyboardSpace={0}
                      onScroll={[Function]}
                      resetKeyboardSpace={[Function]}
                      resetScrollToCoords={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      scrollEventThrottle={1}
                      scrollForExtraHeightOnAndroid={[Function]}
                      scrollIntoView={[Function]}
                      scrollToEnd={[Function]}
                      scrollToFocusedInput={[Function]}
                      scrollToPosition={[Function]}
                      showsVerticalScrollIndicator={true}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "flex": 1,
                        }
                      }
                      update={[Function]}
                      viewIsInsideTabBar={false}
                    >
                      <View>
                        <View>
                          <Styled(Spinner)
                            color="blue"
                          >
                            <Spinner
                              color="blue"
                              style={
                                Object {
                                  "height": 80,
                                }
                              }
                            >
                              <ActivityIndicator
                                animating={true}
                                color="blue"
                                hidesWhenStopped={true}
                                size="large"
                                style={
                                  Object {
                                    "height": 80,
                                  }
                                }
                              >
                                <ActivityIndicator
                                  animating={true}
                                  color="blue"
                                  hidesWhenStopped={true}
                                  size="large"
                                  style={
                                    Object {
                                      "height": 80,
                                    }
                                  }
                                />
                              </ActivityIndicator>
                            </Spinner>
                          </Styled(Spinner)>
                        </View>
                      </View>
                    </RCTScrollView>
                  </_class>
                </ScrollViewMock>
              </KeyboardAwareScrollViewMock>
            </Content>
          </Styled(Content)>
        </View>
      </View>
    </Container>
  </Styled(Container)>
</RedirectorLogic>
`;
